muffnn/mlp/tests/test_mlp_classifier.py

```diff
@@ -357,3 +357,44 @@ def test_sample_weight(make_dataset_func, dataset_kwargs):
         lambda: MLPClassifier(n_epochs=30, random_state=42,
                               keep_prob=0.8, hidden_units=(128,))
     )
+
+
+def test_prediction_gradient():
+    """Test computation of prediction gradients."""
+    # Binary classification
+    n_classes = 1
+    mlp = MLPClassifier(n_epochs=100, random_state=42, hidden_units=(5,))
+    X, y = make_classification(
+        n_samples=1000, n_informative=n_classes, n_redundant=0,
+        n_classes=n_classes, n_clusters_per_class=1, shuffle=False)
```

Comment:
An explicit `n_features=20` might be helpful here. Before I checked the docs I assumed `n_features = n_informative + n_redundant` and was confused.